### PR TITLE
Document Atomic types and objects via comments

### LINKED ISSUE #401: Document `Atomic`, `d.atomic()` and `isAtomicSchema`

### PR DESCRIPTION
**Pull request by mhawryluk**
closes #401

### CODE REVIEW

File: packages/typegpu/src/data/atomic.ts [hunk 1]
```diff
@@ -16,15 +16,34 @@ import type { I32, U32 } from './numeric';
 // Public API
 // ----------
 
+/**
+ * Marks a concrete integer scalar type schema (u32 or i32) as the WGSL atomic.
```
Comment (reczkok) [inline packages/typegpu/src/data/atomic.ts]:
```suggestion
 * Marks a concrete integer scalar type schema (u32 or i32) as a WGSL atomic.
```

**Review by reczkok — APPROVED** [review]
LGTM 🦭 I don't know how specific we want to be with the explanations? Do we assume people should know what an atomic does?

**Comment by mhawryluk** [comment]
@reczkok I was also wondering that. I assume we don't want to duplicate the WGSL documentation though

**Review by iwoplaza — APPROVED** [review]
📖📖📖

**Pull request merged**